api-samples/tabs/inspector/tabs_api.js

```diff
@@ -0,0 +1,403 @@
+  windowItem.querySelector('.window_height').value = window.height;
+  windowItem.querySelector('.window_focused').checked = window.focused;
+  windowItem.querySelector('.window_current').checked = window.current;
+
+  windowItem.querySelector('.window_refresh').addEventListener('click', () => {
+    refreshWindow(window.id);
+  });
+
+  windowItem
+    .querySelector('.update_window_button')
+    .addEventListener('click', () => {
+      updateWindow(window.id);
+    });
+
+  windowItem
+    .querySelector('.remove_window_button')
+    .addEventListener('click', () => {
+      removeWindow(window.id);
+    });
+
+  windowItem
+    .querySelector('.refresh_selected_tab_button')
+    .addEventListener('click', () => {
+      refreshSelectedTab(window.id);
+    });
+
+  windowItem.querySelector('#tabList').innerHTML = '';
+  for (let tab of window.tabs) {
+    const tabItem = document.importNode(tabTemplate, true);
+    renderTab(tab, tabItem.children[0]);
+    windowItem.querySelector('#tabList').appendChild(tabItem);
+  }
+}
+
+function renderTab(tab, tabItem) {
+  tabItem.id = `tab_${tab.id}`;
+  tabItem.querySelector('.tab_index').id = `index_${tab.id}`;
+  tabItem.querySelector('.tab_window_id').id = `windowId_${tab.id}`;
+  tabItem.querySelector('.tab_title').id = `title_${tab.id}`;
+  tabItem.querySelector('.tab_url').id = `url_${tab.id}`;
+  tabItem.querySelector('.tab_selected').id = `selected_${tab.id}`;
+
+  tabItem.querySelector('.tab_id').innerText = `TabId: ${tab.id}`;
+  tabItem.querySelector('.tab_index').value = tab.index;
+  tabItem.querySelector('.tab_window_id').value = tab.windowId;
+  tabItem.querySelector('.tab_title').value = tab.title;
+  tabItem.querySelector('.tab_url').value = tab.url;
+  tabItem.querySelector('.tab_selected').checked = tab.selected;
+
+  tabItem.querySelector('.move_tab_button').addEventListener('click', () => {
+    moveTab(tab.id);
+  });
+  tabItem.querySelector('.refresh_tab_button').addEventListener('click', () => {
+    refreshTab(tab.id);
+  });
+  tabItem.querySelector('.update_tab_button').addEventListener('click', () => {
+    updateTab(tab.id);
+  });
+  tabItem.querySelector('.remove_tab_button').addEventListener('click', () => {
+    removeTab(tab.id);
+  });
+}
+
+function updateTabData(id) {
+  const retval = {
+    url: document.getElementById('url_' + id).value,
+    selected: document.getElementById('selected_' + id).value ? true : false
+  };
+
+  return retval;
+}
+
+async function updateTab(id) {
+  try {
+    await chrome.tabs.update(id, updateTabData(id));
+  } catch (e) {
+    alert(e);
+  }
+}
+
+function moveTabData(id) {
+  return {
+    index: parseInt(document.getElementById('index_' + id).value),
+    windowId: parseInt(document.getElementById('windowId_' + id).value)
+  };
+}
+
+function moveTab(id) {
+  chrome.tabs.move(id, moveTabData(id)).catch(alert);
+}
+
+function createTabData(id) {
+  return {
+    index: parseInt(document.getElementById('index_' + id).value),
```

Comment:
It looks like the HTML fields are wrong here? The HTML has title and url, but we're looking for different things.

Reply:
Oh, this is a negligence. I originally intended to use this function to create and update tabs simultaneously, but after updating the code, I forgot to modify it here. 😥